hypershift-operator/controllers/nodepool/agent.go

```diff
@@ -2,12 +2,19 @@ package nodepool
 
 import (
 	agentv1 "github.com/openshift/cluster-api-provider-agent/api/v1alpha1"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
 )
 
-func agentMachineTemplateSpec() *agentv1.AgentMachineTemplateSpec {
+func agentMachineTemplateSpec(nodePool *hyperv1.NodePool) *agentv1.AgentMachineTemplateSpec {
+	spec := agentv1.AgentMachineSpec{}
```

Comment:
nit: can we just drop the check an inline below?

Reply:
I don't understand what you mean?

Reply:
Just suggesting we drop the check `if nodePool.Spec.Platform.Agent != nil {` and keep this func simple to just return since seems a fair assumption for it to be set when calling this func. Just a thought anyways.